tsmarty2c.php

```diff
@@ -154,34 +180,52 @@ function do_file($outfile, $file) {
 
 	ob_start();
 	echo MSGID_HEADER;
-	foreach($result_msgctxt as $msgctxt => $data_msgid) {
-		foreach($data_msgid as $msgid => $data) {
-			echo "#: ", join(' ', $data['lineno']), "\n";
-			echo 'msgctxt "' . fs($msgctxt) . '"', "\n";
-			echo 'msgid "' . fs($msgid) . '"', "\n";
-			if (isset($data['plural'])) {
-				echo 'msgid_plural "' . fs($data['plural']) . '"', "\n";
-				echo 'msgstr[0] ""', "\n";
-				echo 'msgstr[1] ""', "\n";
-			} else {
-				echo 'msgstr ""', "\n";
-			}
-			echo "\n";
-		}
-	}
-	//without msgctxt
-	foreach($result_msgid as $msgid => $data) {
-		echo "#: ", join(' ', $data['lineno']), "\n";
-		echo 'msgid "' . fs($msgid) . '"', "\n";
-		if (isset($data['plural'])) {
-			echo 'msgid_plural "' . fs($data['plural']) . '"', "\n";
-			echo 'msgstr[0] ""', "\n";
-			echo 'msgstr[1] ""', "\n";
-		} else {
-			echo 'msgstr ""', "\n";
-		}
-		echo "\n";
-	}
+    global $domain;
+    if( isset($domain) && $domain != '' ){
+        foreach($result_msgdomain as $msgdomain => $data_msgid) {
+            foreach($data_msgid as $msgid => $data) {
+                echo "#: ", join(' ', $data['lineno']), "\n";
+                echo 'msgid "' . fs($msgid) . '"', "\n";
+                if (isset($data['plural'])) {
+                    echo 'msgid_plural "' . fs($data['plural']) . '"', "\n";
+                    echo 'msgstr[0] ""', "\n";
+                    echo 'msgstr[1] ""', "\n";
+                } else {
+                    echo 'msgstr ""', "\n";
+                }
+                echo "\n";
+            }
+        }
+    }else{
+        foreach($result_msgctxt as $msgctxt => $data_msgid) {
+            foreach($data_msgid as $msgid => $data) {
+                echo "#: ", join(' ', $data['lineno']), "\n";
+                echo 'msgctxt "' . fs($msgctxt) . '"', "\n";
+                echo 'msgid "' . fs($msgid) . '"', "\n";
+                if (isset($data['plural'])) {
+                    echo 'msgid_plural "' . fs($data['plural']) . '"', "\n";
+                    echo 'msgstr[0] ""', "\n";
+                    echo 'msgstr[1] ""', "\n";
+                } else {
+                    echo 'msgstr ""', "\n";
+                }
+                echo "\n";
+            }
+        }
+        //without msgctxt
+        foreach($result_msgid as $msgid => $data) {
+            echo "#: ", join(' ', $data['lineno']), "\n";
+            echo 'msgid "' . fs($msgid) . '"', "\n";
+            if (isset($data['plural'])) {
+                echo 'msgid_plural "' . fs($data['plural']) . '"', "\n";
+                echo 'msgstr[0] ""', "\n";
+                echo 'msgstr[1] ""', "\n";
+            } else {
+                echo 'msgstr ""', "\n";
+            }
+            echo "\n";
+        }
+    }
```

Comment:
ok, this is too much copy paste to be able to review. i'll try to refactor the code to use more methods rather than continuous code.